# Data gathering and creating a dataset without any NaN values

In [1]:
import requests
import pandas as pd

# Used GET method to grab the data from the TMDB API

In [2]:
response=requests.get('https://api.themoviedb.org/3/movie/popular?api_key=a7e784951966bf44afb76db195d4c9f6&language=en-US&page=1').json()

In [3]:
df=pd.DataFrame()

In [4]:
title=[]
vote_average=[]
popularity=[]
vote_count=[]
original_language=[]
overview=[]
release_date=[]

for i in response['results']:
    print(i,i['vote_average'],i['popularity'],i['vote_count'],i['original_language'],i['overview'],i['release_date'])
    title.append(i['title'])
    vote_average.append(i['vote_average'])
    popularity.append(i['popularity'])
    vote_count.append(i['vote_count'])
    original_language.append(i['original_language'])
    overview.append(i['overview'])
    release_date.append(i['release_date'])
d={'title':title,'vote_average':vote_average,'popularity':popularity,'vote_count':vote_count,'original_language':original_language,'overview':overview,'release_date':release_date}
    
df=pd.DataFrame(d)   

    

{'adult': False, 'backdrop_path': '/qxeqKcVBWnQxUp1w6fwWcxZEA6m.jpg', 'genre_ids': [28, 12, 14], 'id': 436270, 'original_language': 'en', 'original_title': 'Black Adam', 'overview': 'Nearly 5,000 years after he was bestowed with the almighty powers of the Egyptian gods—and imprisoned just as quickly—Black Adam is freed from his earthly tomb, ready to unleash his unique form of justice on the modern world.', 'popularity': 6041.545, 'poster_path': '/3zXceNTtyj5FLjwQXuPvLYK5YYL.jpg', 'release_date': '2022-10-19', 'title': 'Black Adam', 'video': False, 'vote_average': 7.2, 'vote_count': 425} 7.2 6041.545 425 en Nearly 5,000 years after he was bestowed with the almighty powers of the Egyptian gods—and imprisoned just as quickly—Black Adam is freed from his earthly tomb, ready to unleash his unique form of justice on the modern world. 2022-10-19
{'adult': False, 'backdrop_path': '/tIX6j3NzadlwGcJ52nuWdmtOQkg.jpg', 'genre_ids': [27, 53, 9648], 'id': 717728, 'original_language': 'en', 'origina

# Created a DataFrame with the data inside 'd' dictionary that i got from the API's frist page

In [5]:
df

title  vote_average  popularity  \
0                                       Black Adam           7.2    6041.545   
1                         Jeepers Creepers: Reborn           5.7    5365.773   
2                                   Halloween Ends           6.7    3957.709   
3           Fullmetal Alchemist: The Final Alchemy           6.4    3476.600   
4                               Orphan: First Kill           6.8    2955.222   
5                                             Fall           7.3    2483.067   
6                                   Project Gemini           5.5    2448.593   
7                                     Bullet Train           7.5    2412.858   
8                                      Terrifier 2           7.7    2357.513   
9   The Battle at Lake Changjin: Water Gate Bridge           6.3    2352.585   
10                                       Grimcutty           5.8    2346.774   
11                                          Athena           6.6    2186.455   
12                                   Hocus Pocus 2           7.6    2111.004   
13                            The Infernal Machine           7.0    2076.136   
14                             Secret Headquarters           6.8    2058.629   
15                                        Blackout           5.5    1888.472   
16                                       Pinocchio           6.7    1581.168   
17                                       Terrifier           6.5    1524.631   
18                                           Beast           7.0    1493.128   
19                                             Lou           6.6    1457.168   

    vote_count original_language  \
0          425                en   
1          333                en   
2          628                en   
3           98                ja   
4         1103                en   
5         1526                en   
6          129                en   
7         1969                en   
8           51                en   
9           19                zh   
10         152                en   
11         303                fr   
12         863                en   
13          91                en   
14         122                en   
15          52                en   
16         969                en   
17         700                en   
18         706                en   
19         307                en   

                                             overview release_date  
0   Nearly 5,000 years after he was bestowed with ...   2022-10-19  
1   Forced to travel with her boyfriend to a horro...   2022-09-15  
2   Four years after the events of Halloween in 20...   2022-10-12  
3   The Elric brothers’ long and winding journey c...   2022-06-24  
4   After escaping from an Estonian psychiatric fa...   2022-07-27  
5   For best friends Becky and Hunter, life is all...   2022-08-11  
6   After depleting Earth's resources for centurie...   2022-01-06  
7   Unlucky assassin Ladybug is determined to do h...   2022-07-03  
8   After being resurrected by a sinister entity, ...   2022-10-06  
9   In the follow-up to "The Battle At Lake Changj...   2022-02-01  
10  A suburban teen girl and her little brother mu...   2022-10-10  
11  Hours after the tragic death of their youngest...   2022-09-09  
12  It's been 29 years since someone lit the Black...   2022-09-27  
13  Reclusive and controversial author Bruce Cogbu...   2022-09-23  
14  While hanging out after school, Charlie and hi...   2022-08-12  
15  A man wakes in a hospital with no memory, and ...   2022-10-12  
16  A wooden puppet embarks on a thrilling adventu...   2022-09-07  
17  On Halloween night, a young woman finds hersel...   2016-10-15  
18  A recently widowed man and his two teenage dau...   2022-08-11  
19  A young girl is kidnapped during a powerful st...   2022-09-23

# Complete programme for cretaing data set from TMDB movie database which is of 500 json pages when we view it from jason online viewer
# Using for loop to iterate through all the pages and append all the data to the corresponding empty list


In [6]:
import requests
import pandas as pd

In [7]:
movie_dataset=pd.DataFrame()
for j in range(1,500):
    response=requests.get('https://api.themoviedb.org/3/movie/popular?api_key=a7e784951966bf44afb76db195d4c9f6&language=en-US&page={}'.format(j)).json()
    title=[]
    vote_average=[]
    popularity=[]
    vote_count=[]
    original_language=[]
    overview=[]
    release_date=[]
    for i in response['results']:
        try:
            release_date.append(i['release_date'])
        except KeyError:
            continue 

        title.append(i['title'])
        vote_average.append(i['vote_average'])
        popularity.append(i['popularity'])
        vote_count.append(i['vote_count'])
        original_language.append(i['original_language'])
        overview.append(i['overview'])
        release_date.append(i['release_date'])
        
    
    d={'title':title,'vote_average':vote_average,'popularity':popularity,'vote_count':vote_count,'original_language':original_language,'overview':overview,'release_date':release_date}
    d = dict([ (k,pd.Series(v)) for k,v in d.items() ])
    df=pd.DataFrame(d) 
    movie_dataset=pd.concat([df,movie_dataset],ignore_index=True)
 

In [8]:
   movie_dataset

title  vote_average  popularity  vote_count  \
0                   Slate           5.3      11.813         9.0   
1          Days of Heaven           7.5      11.972       800.0   
2           Dear Dictator           5.6      11.812       136.0   
3      Dangerous Liaisons           5.8      11.807       193.0   
4                Scanners           6.7      10.221       927.0   
...                   ...           ...         ...         ...   
19933                 NaN           NaN         NaN         NaN   
19934                 NaN           NaN         NaN         NaN   
19935                 NaN           NaN         NaN         NaN   
19936                 NaN           NaN         NaN         NaN   
19937                 NaN           NaN         NaN         NaN   

      original_language                                           overview  \
0                    ko  Yeon-hee always had dreams of becoming a heroi...   
1                    en  In 1916, a Chicago steel worker accidentally k...   
2                    en  When political turmoil forces a British-Caribb...   
3                    fr  The innocent Célène might be falling in love w...   
4                    en  After a man with extraordinary—and frightening...   
...                 ...                                                ...   
19933               NaN                                                NaN   
19934               NaN                                                NaN   
19935               NaN                                                NaN   
19936               NaN                                                NaN   
19937               NaN                                                NaN   

      release_date  
0       2020-10-24  
1       2020-10-24  
2       1978-09-13  
3       1978-09-13  
4       2018-03-16  
...            ...  
19933   2016-10-15  
19934   2022-08-11  
19935   2022-08-11  
19936   2022-09-23  
19937   2022-09-23  

[19938 rows x 7 columns]

In [9]:
movie_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19938 entries, 0 to 19937
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   title              9969 non-null   object 
 1   vote_average       9969 non-null   float64
 2   popularity         9969 non-null   float64
 3   vote_count         9969 non-null   float64
 4   original_language  9969 non-null   object 
 5   overview           9969 non-null   object 
 6   release_date       19938 non-null  object 
dtypes: float64(3), object(4)
memory usage: 1.1+ MB


In [10]:
movie_dataset.isna().sum()

title                9969
vote_average         9969
popularity           9969
vote_count           9969
original_language    9969
overview             9969
release_date            0
dtype: int64

# Showing the first 10 rows of the raw datset

In [11]:
movie_dataset.head(10)

title  vote_average  popularity  vote_count original_language  \
0               Slate           5.3      11.813         9.0                ko   
1      Days of Heaven           7.5      11.972       800.0                en   
2       Dear Dictator           5.6      11.812       136.0                en   
3  Dangerous Liaisons           5.8      11.807       193.0                fr   
4            Scanners           6.7      10.221       927.0                en   
5            The Hole           6.2      14.272       684.0                en   
6       The Iron Lady           6.4      14.224      1486.0                en   
7         Up in Smoke           6.8       9.771       463.0                en   
8                They           4.9      11.522       205.0                en   
9       Stand Up Guys           6.0      11.804       776.0                en   

                                            overview release_date  
0  Yeon-hee always had dreams of becoming a heroi...   2020-10-24  
1  In 1916, a Chicago steel worker accidentally k...   2020-10-24  
2  When political turmoil forces a British-Caribb...   1978-09-13  
3  The innocent Célène might be falling in love w...   1978-09-13  
4  After a man with extraordinary—and frightening...   2018-03-16  
5  Four teenagers at a British private school sec...   2018-03-16  
6  A look at the life of Margaret Thatcher, the f...   2022-07-08  
7  An unemployed pot-smoking slacker and amateur ...   2022-07-08  
8  After witnessing a horrific and traumatic even...   1981-01-14  
9  After serving 28 years in prison for accidenta...   1981-01-14

# Showing the bottom 10 rows using tail()

In [12]:
 movie_dataset.tail(10)

title  vote_average  popularity  vote_count original_language overview  \
19928   NaN           NaN         NaN         NaN               NaN      NaN   
19929   NaN           NaN         NaN         NaN               NaN      NaN   
19930   NaN           NaN         NaN         NaN               NaN      NaN   
19931   NaN           NaN         NaN         NaN               NaN      NaN   
19932   NaN           NaN         NaN         NaN               NaN      NaN   
19933   NaN           NaN         NaN         NaN               NaN      NaN   
19934   NaN           NaN         NaN         NaN               NaN      NaN   
19935   NaN           NaN         NaN         NaN               NaN      NaN   
19936   NaN           NaN         NaN         NaN               NaN      NaN   
19937   NaN           NaN         NaN         NaN               NaN      NaN   

      release_date  
19928   2022-10-12  
19929   2022-10-12  
19930   2022-09-07  
19931   2022-09-07  
19932   2016-10-15  
19933   2016-10-15  
19934   2022-08-11  
19935   2022-08-11  
19936   2022-09-23  
19937   2022-09-23

# Checking the shape

In [13]:
 movie_dataset.shape

(19938, 7)

#  Counted the no of NaN values in raw dataset

In [14]:
count_nan =  movie_dataset.isna().sum()

In [15]:
count_nan

title                9969
vote_average         9969
popularity           9969
vote_count           9969
original_language    9969
overview             9969
release_date            0
dtype: int64

# we found out that there are so many NaN values .
# Now by using dropna() removing all the rows with NaN

In [16]:
new_movie_dataset= movie_dataset.dropna()

# we got the new movie data frame but the index is not in order
# I used reset_index () to set the order of index 

In [17]:
new_movie_dataset

title  vote_average  popularity  vote_count  \
0                   Slate           5.3      11.813         9.0   
1          Days of Heaven           7.5      11.972       800.0   
2           Dear Dictator           5.6      11.812       136.0   
3      Dangerous Liaisons           5.8      11.807       193.0   
4                Scanners           6.7      10.221       927.0   
...                   ...           ...         ...         ...   
19913            Blackout           5.5    1888.472        52.0   
19914           Pinocchio           6.7    1581.168       969.0   
19915           Terrifier           6.5    1524.631       700.0   
19916               Beast           7.0    1493.128       706.0   
19917                 Lou           6.6    1457.168       307.0   

      original_language                                           overview  \
0                    ko  Yeon-hee always had dreams of becoming a heroi...   
1                    en  In 1916, a Chicago steel worker accidentally k...   
2                    en  When political turmoil forces a British-Caribb...   
3                    fr  The innocent Célène might be falling in love w...   
4                    en  After a man with extraordinary—and frightening...   
...                 ...                                                ...   
19913                en  A man wakes in a hospital with no memory, and ...   
19914                en  A wooden puppet embarks on a thrilling adventu...   
19915                en  On Halloween night, a young woman finds hersel...   
19916                en  A recently widowed man and his two teenage dau...   
19917                en  A young girl is kidnapped during a powerful st...   

      release_date  
0       2020-10-24  
1       2020-10-24  
2       1978-09-13  
3       1978-09-13  
4       2018-03-16  
...            ...  
19913   2022-07-03  
19914   2022-10-06  
19915   2022-10-06  
19916   2022-02-01  
19917   2022-02-01  

[9969 rows x 7 columns]

In [18]:
new_movie_dataset.shape

(9969, 7)

In [19]:
new_movie_dataset.reset_index(inplace=True, drop=True)

In [20]:
new_movie_dataset

title  vote_average  popularity  vote_count  \
0                  Slate           5.3      11.813         9.0   
1         Days of Heaven           7.5      11.972       800.0   
2          Dear Dictator           5.6      11.812       136.0   
3     Dangerous Liaisons           5.8      11.807       193.0   
4               Scanners           6.7      10.221       927.0   
...                  ...           ...         ...         ...   
9964            Blackout           5.5    1888.472        52.0   
9965           Pinocchio           6.7    1581.168       969.0   
9966           Terrifier           6.5    1524.631       700.0   
9967               Beast           7.0    1493.128       706.0   
9968                 Lou           6.6    1457.168       307.0   

     original_language                                           overview  \
0                   ko  Yeon-hee always had dreams of becoming a heroi...   
1                   en  In 1916, a Chicago steel worker accidentally k...   
2                   en  When political turmoil forces a British-Caribb...   
3                   fr  The innocent Célène might be falling in love w...   
4                   en  After a man with extraordinary—and frightening...   
...                ...                                                ...   
9964                en  A man wakes in a hospital with no memory, and ...   
9965                en  A wooden puppet embarks on a thrilling adventu...   
9966                en  On Halloween night, a young woman finds hersel...   
9967                en  A recently widowed man and his two teenage dau...   
9968                en  A young girl is kidnapped during a powerful st...   

     release_date  
0      2020-10-24  
1      2020-10-24  
2      1978-09-13  
3      1978-09-13  
4      2018-03-16  
...           ...  
9964   2022-07-03  
9965   2022-10-06  
9966   2022-10-06  
9967   2022-02-01  
9968   2022-02-01  

[9969 rows x 7 columns]

In [21]:
new_movie_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9969 entries, 0 to 9968
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   title              9969 non-null   object 
 1   vote_average       9969 non-null   float64
 2   popularity         9969 non-null   float64
 3   vote_count         9969 non-null   float64
 4   original_language  9969 non-null   object 
 5   overview           9969 non-null   object 
 6   release_date       9969 non-null   object 
dtypes: float64(3), object(4)
memory usage: 545.3+ KB


In [22]:
new_movie_dataset.isna().sum()

title                0
vote_average         0
popularity           0
vote_count           0
original_language    0
overview             0
release_date         0
dtype: int64

# Finally created a meaningful Data set with 9969 rows and 7 columns!